autogen/oai/completion.py

```diff
@@ -203,10 +208,13 @@ def _get_response(cls, config: Dict, raise_on_ratelimit_or_timeout=False, use_ca
         retry_wait_time = config.pop("retry_wait_time", cls.retry_wait_time)
         while True:
             try:
-                if "request_timeout" in config:
-                    response = openai_completion.create(**config)
+                if "request_timeout" not in config:
+                    config["request_timeout"] = request_timeout
+                api_type = config.get("api_type", None)
+                if api_type and re.sub(r'[^a-zA-Z0-9]', '', api_type).lower() == "litellm":
+                    response = litellm.completion(**config)
```

Comment:
I think some sanitizations should be done on the `config` dict.
Now, it's sending `api_type = 'litellm'` down to `litellm.completion` which doesn't sanitize it's inputs to each llm service.

For example, Anthropic's Claude API doesn't work due to:
```
litellm.exceptions.InvalidRequestError: AnthropicException - {'type': 'invalid_request_error', 'message': 'api_type: extra fields not permitted'}
```

Reply:
```diff
Subject: [PATCH] sanitize config before passing to litellm
---
Index: autogen/oai/completion.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/autogen/oai/completion.py b/autogen/oai/completion.py
--- a/autogen/oai/completion.py	(revision b54240d1d7b94f78a18d9b6766d680aaaa13391e)
+++ b/autogen/oai/completion.py	(date 1696848902124)
@@ -212,7 +212,9 @@
                     config["request_timeout"] = request_timeout
                 api_type = config.get("api_type", None)
                 if api_type and re.sub(r'[^a-zA-Z0-9]', '', api_type).lower() == "litellm":
-                    response = litellm.completion(**config)
+                    sanitized_config = config.copy()
+                    del sanitized_config["api_type"]
+                    response = litellm.completion(**sanitized_config)
                 else:
                     response = openai_completion.create(**config)
             except (
```

Reply:
Good catch. Thanks!